### DISCUSSION TITLE
如何修改balbel的配置，需要修改compact参数的配置

### DISCUSSION THREAD

**Discussion by starsoul666** [umijs/umi#11508] [Q&A]
umi支持覆盖babel的配置吗，如何把babel的compact参数设置为false

{
  "compact": false
}

**Comment by fz6m** [comment]
umi 4 的修改方法：

```ts
// .umirc.ts

  chainWebpack(config) {
    ;['src', 'jsx-ts-tsx', 'extra-src'].forEach(rule => {
      config.module
        .rule(rule).use('babel-loader').tap(opts => {
          opts.compact = false
          return opts
        })
    })
    return config
  },
```

**Comment by starsoul666** [comment]
umi3.0的如何配置



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2023年08月10日 21:03 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [umijs/umi] 如何修改balbel的配置，需要修改compact参数的配置 (Discussion #11508) |

umi 4 的修改方法：

// .umirc.tschainWebpack(config){;['src','jsx-ts-tsx','extra-src'].forEach(rule=>{config.module.rule(rule).use('babel-loader').tap(opts=>{opts.compact=falsereturnopts})})returnconfig},

**Reply by fz6m** [reply]
umi 3 不维护了，建议升级到 umi 4
